### PR TITLE
Move publish pipeline to private 1ES Agent pools

### PR DESCRIPTION
**Pull request by dannyvv**
null

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8466)

### CODE REVIEW

**Review by NickGerleman — APPROVED** [review]
LGTM.

If I recall, we had a specific workaround somewhere where we needed to replace a task name with a guid for usage in the Microsoft pool. We can probably clean that up now. @jonthysell would have details I think.

**Comment by dannyvv** [comment]
Confirmed with a main-branch run that all steps seem to run fine, so started backports of 0.65 and 0.64

File: .ado/publish.yml [hunk 1]
```diff
@@ -40,15 +40,18 @@ variables:
...
 
 trigger: none
 pr: none
 
 jobs:
   - job: RnwNpmPublish
     displayName: React-Native-Windows Npm Build Rev Publish
-    pool:
-      vmImage: $(VmImage)
+    pool: $(AgentPool.Medium.Microsoft)
```
Comment (JunielKatarn) [inline .ado/publish.yml:54]: Why use a new variable name instead of having different values to choose from for `$(VmImage)`?

Comment (dannyvv) [reply .ado/publish.yml:54]: We haven't imported those into the publish template yet, so I didn't want to start now.
I have hopes that some time I have time to improve the publish pipeline by splitting out the build part with the CI pipeline's build outputs and then have the release definitoin take care of the git push, npm push and nuget pushes...

**Pull request merged**